ipi/utils/instools.py

```diff
@@ -5,39 +5,80 @@
 import ipi.utils.mathtools as mt
 import os.path
 import glob
+import copy
```

Comment:
this came from the merge... not sure what happened but can't hurt too much. I will accept this pull request and hope for the best.